python/lsst/ap/verify/dataset.py

```diff
@@ -161,20 +162,43 @@ def configLocation(self):
     def obsPackage(self):
         """The name of the obs package associated with this dataset (`str`, read-only).
         """
-        return Butler.getMapperClass(self._stubInputRepo).getPackageName()
+        return dafPersistence.Butler.getMapperClass(self._stubInputRepo).getPackageName()
 
     @property
     def camera(self):
-        """The name of the camera associated with this dataset (`str`, read-only).
+        """The name of the Gen 2 camera associated with this dataset (`str`, read-only).
         """
-        return Butler.getMapperClass(self._stubInputRepo).getCameraName()
+        return dafPersistence.Butler.getMapperClass(self._stubInputRepo).getCameraName()
+
+    @property
+    def instrument(self):
+        """The name of the Gen 3 instrument associated with this dataset (`str`, read-only).
```

Comment:
This use of the term "dataset" is at odds with the gen3 use of "dataset". We use dataset to mean a single thing in registry. From that standpoint, it's very confusing to read this and not be confused. A dataset in gen3 by definition only can be associated with one instrument.  This method is trying to determine if the entire butler repository only has one instrument registered, which is very different.